firmware/mcal/stm32f767/os/timer.h

```diff
@@ -5,33 +5,33 @@
 
 #include <cstdint>
 
-#include "shared/util/os.h"
-#include "shared/os/timer.h"
 #include "cmsis_os2.h"
+#include "shared/os/os.h"
+#include "shared/os/timer.h"
 
 namespace mcal::os {
 
-
 // TODO: Add comments and handle errors more robustly
-class Timer final : public shared::os::Timer  {
-private:
-    osTimerId_t* timer_id;
+class Timer final : public shared::os::Timer {
```

Comment:
based on discussion, change to `SoftTimer`

Reply:
SoftTimer as in SoftwareTimer? If so can I use the full word to make it more clear?

Reply:
I realized that the name is already quite descriptive since the namespace makes it `os::Timer` which perfectly describes its origin and purpose

Reply:
makes perfect sense. no change needed